models/dataset.go

```diff
@@ -507,32 +498,15 @@ func ValidateDataset(ctx context.Context, dataset *Dataset) error {
 
 }
 
-func santisedGeneralDetails(input []GeneralDetails) ([]GeneralDetails, []string) {
-	sanitisedDetails := make([]GeneralDetails, 0)
-	invalidFields := make([]string, 0)
-
-	for _, original := range input {
-		cleanHref := strings.TrimSpace(original.HRef)
-		_, err := url.Parse(cleanHref)
+func validateGeneralDetails(generalDetails []GeneralDetails, identifier string) (invalidFields []string) {
+	for i, gd := range generalDetails {
+		_, err := url.Parse(gd.HRef)
 		if err != nil {
-			invalidFields = append(invalidFields, "Href")
+			invalidFields = append(invalidFields, fmt.Sprintf("%s[%d].HRef", identifier, i))
```

Comment:
The unit test code does not get to line 505.
It would be good if a unit test was added that caused the Parse of the HRef on line 503 to return an error.

Reply:
I see the code has been changed to add test for this line